### DISCUSSION TITLE
How to get result of the executed patch command from handleModelUpdate() method in ModelserverAwareWidgetProvider

### DISCUSSION THREAD

**Discussion by quratulain-ali** [eclipse-glsp/glsp#1206] [General] [answered]
The argument `modelUpdate: IncrementalUpdateNotificationV2` in method `handleModelUpdate()` contains the patch which needs to be executed to get the resultant updated model. Is there a way we can access the updated model or the result of the patch command?

**package** `@eclipse-emfcloud/modelserver-jsonforms-property-view`
**class**       `ModelserverAwareWidgetProvider`
**method**  `protected abstract handleModelUpdate(modelUpdate: IncrementalUpdateNotificationV2, semanticUri: string): void;`

**Comment by ndoschek** [comment, marked as answer]
Hi @quratulain-ali!
Thanks for your reaching out!

If you use the `ModelServerAware*` version, there is currently no other option, than the provided incremental update patch.
However, you could query the current model within the `handleModelUpdate` method via the `modelServerClient` member.
This would ensure that you get the latest updated version of the model, but would of course mean another server roundtrip.

HTH, best wishes,
Nina

**Reply by quratulain-ali** [reply]
Thank you for your response. To avoid another server roundtrip, I found a workaround to call `updateContentWidget` with the selected element id to update the property view data
